system/lib/wasmfs/syscalls.cpp

```diff
@@ -1492,8 +1492,11 @@ int __syscall_fcntl64(int fd, int cmd, ...) {
       flags = va_arg(v1, int);
       va_end(v1);
       // This syscall should ignore most flags.
+      // We are adding nonblock as we are running in nonblocking
+      // by default, but some libraries and apps still want to set it,
+      // so we allow it
       flags = flags & ~(O_RDONLY | O_WRONLY | O_RDWR | O_CREAT | O_EXCL |
-                        O_NOCTTY | O_TRUNC);
+                        O_NOCTTY | O_TRUNC | O_NONBLOCK);
```

Comment:
Rather than ignoring `O_NONBLOCK` wouldn't we want to pass it through to setFlags?   i.e. shouldn't we actually support it?

Reply:
Hi Sam,

From what I have read, I think it's supported essentially by default, or rather, blocking is not supported.

Reply:
I don't think this is true, IIUC all our filesystem support is blocking, which is achieved either by preloading the files into memory or using a background thread to perform async operations.  

IIUC we don't currently support O_NONBLOCK, although it would probably very logical to do so.

@tlively ?

Reply:
I think the flag needs to go through the backend, not be ignored here.  If the FS supports NON_BLOCKING (as in pipes), and we should accept the flag.  For filesystems that don't currently support NON_BLOCKING we should probably reject it.

Reply:
Hi Sam,

I agree.  I'm working on that piece now; I need to get up to speed a little bit with the code, but I'll modify this PR when I have a fix.

Thanks for your help and recommendation!

-Andrew

Reply:
@sbc100 Hi Sam, I've made some changes to the backend and passed the flag into the backend.  Just bumping.

-Andrew

Reply:
Emscripten pipes are weird because they are non-blocking by default whereas native pipes are blocking by default. I agree we should do something reasonable with O_NONBLOCK and we should let backends support it in other cases as well.

Reply:
@tlively @sbc100  My change adds the ability for specific file types to block specific flags, such as O_NONBLOCK.  Right now, it always returns true for everything, but could you go into more detail for me?  What I see with the "Pipe" file type is that it simply writes the input to the output buffer, the one that is read by the other end.  I had a gander through the other file types that we currently support and didn't notice any abnormalities.

The behaviour is symmetrical with what we had before, at least with O_APPEND, as there is nowhere that it is checked to see if it can actually be done.  Is this correct?